llvm/lib/Target/RISCV/RISCVFeatures.td

```diff
@@ -1359,6 +1359,10 @@ def TuneOptimizedZeroStrideLoad
                       "true", "Optimized (perform fewer memory operations)"
                       "zero-stride vector load">;
 
+def TuneOptimizedIndexedLoadStore
+   : SubtargetFeature<"optimized-indexed-load-store", "HasOptimizedIndexedLoadStore",
```

Comment:
Its unclear what "optimized" means here.

Reply:
`fast-indexed-load-store`?

Reply:
Its not clear how fast or optimized it needs to be. How would a future CPU know they should set this? It doesn't say what it does.

Reply:
I feel like I may have derailed this PR with my vluxeiN.v suggestion earlier, sorry. If we want to defer the decision about the subtarget feature till later I'm happy if we only do the vrgather.vv lowering in this PR. Especially given that according to my last review comment it does seem to be more performant than vluxeiN.v on the BPI F3

Reply:
I'm also happy with only vrgather.vv lowering

Reply:
Done! We can come back to add the indexed load way later and please review the latest patch. :-)